index.bs

```diff
@@ -181,96 +182,66 @@ partial interface Navigator {
 
   1. [=Enumerate XR devices=].
-  1. If the [=list of XR devices=] is empty, [=reject=] |promise| with a {{NotFoundError}} and abort these steps.
-  1. [=Select a default XR device=] from the [=list of XR devices=].
-  1. [=/Resolve=] |promise| with the [=default device=].
+  1. [=Select an XR device=] from the [=list of XR devices=].
 
 </div>
 
-Calling {{XR/requestDevice()}} MUST NOT trigger device-selection UI as this would cause many sites to display XR-specific dialogs early in the document lifecycle without user activation.
-
-ISSUE: Take permissions into account when calling {{XR/requestDevice}}.
-
 The <dfn attribute for="XR">ondevicechange</dfn> attribute is an [=Event handler IDL attribute=] for the {{devicechange}} event type.
 
-<div class="example">
-The following code finds an available {{XRDevice}}.
-
-<pre highlight="js">
-navigator.xr.requestDevice().then(device => {
-  // Resolves if an XRDevice is available.
-  onXRAvailable(device);
-}).catch(error => {
-  // An error occurred while requesting an XRDevice or none is available.
-  console.error('Unable to retrieve an XR device: ', error);
-});
-</pre>
-</div>
-
-
-XRDevice {#xrdevice-interface}
----------
-
-<pre class="idl">
-[SecureContext, Exposed=Window] interface XRDevice {
-  // Methods
-  Promise&lt;void&gt; supportsSession(optional XRSessionCreationOptions options);
-  Promise&lt;XRSession&gt; requestSession(optional XRSessionCreationOptions options);
-};
-</pre>
-
-An {{XRDevice}} represents a physical unit of XR hardware that can present imagery to the user somehow. On desktop devices this may take the form of a headset peripheral; on mobile devices it may represent the device itself in conjunction with a viewer harness. It may also represent devices without the ability to present content in stereo but with advanced (6DoF) tracking capabilities.
-
-Each {{XRDevice}} has a <dfn>supports immersive</dfn> value, which is a boolean which MUST be set to <code>true</code> if the device can support [=immersive sessions=] and <code>false</code> if it cannot.
-
-Each {{XRDevice}} has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of non-immersive sessions</dfn>, which MUST be initially empty.
+Each [=XR device=] has a <dfn>supports immersive</dfn> value, which is a boolean which MUST be set to <code>true</code> if the device can support [=immersive sessions=] and <code>false</code> if it cannot.
 
-<div class="algorithm" data-algorithm="supports-session">
+<div class="algorithm" data-algorithm="supports-session-mode">
 
-When the <dfn method for="XRDevice">supportsSession(|options|)</dfn> method is invoked, it MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+When the <dfn method for="XR">supportsSession(|options|)</dfn> method is invoked, it MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
```

Comment:
Is this really "in parallel"?

Reply:
It's a promise, so yes.

Reply:
But each of the steps aren't in parallel, yeah?